prime-router/docs/proposals/0017-metadata-tables-cleanup.md

```diff
@@ -0,0 +1,441 @@
+
+1. Create a new independent queue in distinct namespace that will hold the raw messages to be processed
+2. Create a separate Azure function that will accept the message and process the contents of the queue
+
+### Discussion
+By implementing these changes, repeated information like facility details, provider information, as well as geographic information can be cleaned up and more easily indexed. In addition, we are able to collect more complex information in this structure, for example a message having multiple AOE questions, or multiple specimens, or multiple ordered tests. We also can begin to build in more advanced data verification, such as checking CLIAs or NPIs. We also have the option to extend out features like our address enrichment by adding things like census tract details and ZIP+4.  
+
+### Link to the proposed changes
+You can review the currently proposed SQL changes here, or at the bottom in Appendix 1:
+
+
+## Further Work
+Once feedback has been provided about the structure of the tables and the plan above, any proposed changes will go through an acceptance/rejection process, and then the actual changes will be put into a PR for a final review. This document has been created on 3 November 2022, and will be reviewed at the developer sync on Monday, 7 November 2022 with a targeted proposal acceptance date of 14 November 2022. At that point the work will begin on Phase 1 with an expected completion date of 21 November 2022. Start and completion date for Phase 2 TBD.
+
+It should also be noted that this work is not to make a system of record for the CDC or for our STLT partners. We are still only acting as a hub for the transferring of data. That said, having the ability to provide usage and demographic data to our senders and receivers means we provide a richer and better experience for our users, and it empowers our support staff.
+
+## Appendix 1
+
+```postgresql
+/**
+    Changes to the metadata tables, unifying them to a single normalized view
+    Part of proposal 0017-metadata-table-cleanup
+    Maurice Reeves - qva8@cdc.gov
+    3 November 2022
+*/
+
+-- create a country table
+create table country
+(
+    country_id                  bigserial not null primary key
+    , country_name              varchar(512) not null
+    , country_abbreviation      varchar(50) not null
+);
+
+comment on table country is
+    'A table of countries for the messages we get. These should correspond to the ISO-3166 list';
+comment on column country.country_id is
+    'The primary key for the country table';
+comment on column country.country_name is
+    'The name of the country per ISO-3166, not the official state name value';
+comment on column country.country_abbreviation is
+    'This the alpha-3 code value from HL70399 or ISO-3166, for example USA not US';
+
```

Comment:
Is timezone auto generated?

Reply:
We are trying to keep all date times in UTC in the DB so we don’t have to worry about time zone.